### ISSUE TITLE
deprecate `ignoreUncleanSSLShutdown: Bool` parameter 

### ISSUE THREAD

**Issue by weissi** [swift-server/async-http-client#231]
We should probably deprecate the `ignoreUncleanSSLShutdown` parameter. AsyncHTTPClient _knows_ when it can and when it cannot ignore an unclean SSL shutdown.

Basically if an `.end` was received and we have either a `content-length` OR a `transfer-encoding: chunked` header from the server, we can safely ignore. For HTTP/2 we can always ignore.

If we do have neither a `content-length`, nor a `transfer-encoding: chunked` header, then this should error.

**Comment by weissi** [comment]
@Lukasa does this sound right?

**Comment by Lukasa** [comment]
_Probably_, with the caveat that in principle we may want users to be able to tolerate even the last case if they're sure they should. But frankly I think deprecating sounds good.

**Comment by weissi** [comment]
I agree with that. My thinking was that EOF framing over TLS should be super rare (am I right here?), and even rarer without CloseNotify, so I don't think that's a thing we need to support. And if we do then we can add a new one with a more scary name :). `ignoreTrulyInsecureUncleanTLSShutdowns: Bool` :)

**Comment by Lukasa** [comment]
I believe it should be super rare, yes. As noted, I’m happy to deprecate.

**Comment by weissi** [comment]
For more context: #238 contains the full explanation why we don't need this parameter.
